src/Conjure/Language/Expression/Op/LexLeq.hs

```diff
@@ -25,7 +25,7 @@ instance (TypeOf x, Pretty x) => TypeOf (OpLexLeq x) where
     typeOf p@(OpLexLeq a b) = do
         tyA <- typeOf a
         tyB <- typeOf b
-        if typesUnify [TypeList TypeAny, tyA, tyB]
```

Comment:
I wonder why the [?] thing was here. Maybe it allows <lex on comprehensions? Why did you have to remove it?

Reply:
It just restricts lex ops to work on comprehensions - I have a type check after flattenLex to enforce 1D matrix or list of Int. It could be moved here I suppose, should have the same effect.

Reply:
Looking at this a bit more... The reason I remove the TypeList is because after symmetryOrdering when we introduce <lex or <=lex we can have either a tuple or a list or a matrix either side. So the unification relaxation is there until we get to the flattenLex rule which enforces 1D.